packages/driver/src/cy/multi-domain/commands_manager.ts

```diff
@@ -57,15 +58,21 @@ export class CommandsManager {
     Cypress.action('cy:enqueue:command', attrs)
   }
 
-  endCommand = ({ id, name, err, logId }) => {
+  endCommand = ({ id, subject, failedToSerializeSubjectOfType, name, err, logId }) => {
     const command = this.commands[id]
 
     if (!command) return
 
     delete this.commands[id]
 
     if (!err) {
-      return command.deferred.resolve()
+      if (failedToSerializeSubjectOfType) {
+        return command.deferred.resolve(
+          failedToSerializeSubject(failedToSerializeSubjectOfType),
+        )
+      }
```

Comment:
this logic, as minimal as it is, is duplicated  in index.ts (see below). Is it worth trying to do some special handling of the end:command and ran:domain:fn events in the communicator to swap in the proxy for the subject before these commands fire? (I think that could work?)